### DISCUSSION TITLE
One instance unlinked and signal cli shows an error

### DISCUSSION THREAD

**Discussion by thlkmoeller** [AsamK/signal-cli#1198] [Q&A]
Hello,

I am using Linux Mint on 3 laptops and I don't have a smartphone. I created an account for Signal with Signal-cli with my mobile number on one of those laptops. Then I linked 3 instances of Signal-Desktop with it, each on one of the laptops. The "main" instance is on one laptop. Unfortunately this laptop I could't use for about 5 weeks. Now I started it again and the Signal-Desktop app shows me, that it is not coupled anymore and that I should repeat the coupling process. But the signal-cli instance, which i used to couple the laptops (on the same laptop), shows me an error now:

"
thl@thl-ThinkPad-X200:~/Downloads/signal-cli-0.7.4/bin$ ./signal-cli listDevices
ERROR App - Error while checking account +49XXXXXXXXXX: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
"

The two other instances of Signal Desktop still work fine.

Do You have any idea, how I can reconnect the 3rd instance? And what happened to signal cli? what is this error and how can i fix it? Can someone please help?

**Comment by AsamK** [comment]
You need to update signal-cli or replace the certificate store (https://github.com/AsamK/signal-cli/issues/1067#issuecomment-1295204655)

**Reply by thlkmoeller** [reply]
Hello, thanks for the fast answer. Unfortunately I cannot get it done. I have no lib.jar in the /lib directory. See screenshot attached. It is version 0.7.4
Updating to a new version also didn't work for me. Attached another screenshot. How anyway would the new version (in the new subfolder) "know", that my phone number is registered?

![Bildschirmfoto_2023-03-08_14-03-39](https://user-images.githubusercontent.com/77677066/223720469-c7736686-41bc-4406-9eaf-ce23531fe7f2.png)
![Bildschirmfoto_2023-03-08_14-03-00](https://user-images.githubusercontent.com/77677066/223720478-429eef49-24c9-414f-a29c-2debb57fabb0.png)

**Reply by AsamK** [reply]
Ah, then the version is even older, and you'd need to replace it in the signal-cli-*.jar file.

The error message of the new version means that you need a newer java version, at least java 17.

The account data is stored in a user specific folder, not in the program directory.
Though you might need to move the directory to its new folder, see https://github.com/AsamK/signal-cli/blob/master/CHANGELOG.md#breaking-changes-1

**Reply by thlkmoeller** [reply]
Hello,

thank You!
I made at least signal-cli work again, although it was a little tricky.
I used to zip editor to access the file signal-cli-0.7.4.jar.
There I found the file whisper.store in the following path:
/org/asamk/signal/manager/

I deleted it and replaced it with the newly downloaded one.

Then I went to the command prompt and entered (I use Linux Mint):

./signal-cli listDevices

Interestingly all devices are listed, which means:

device 1 (this device) is the signal-cli instance.

and the devices 2 till 4 are the signal desktop devices.
All devices are listed with a long name (except for device 1, signal-cli), the date, when they were created and the date, when they were "last seen".

But the device, which is supposedly not linked anymore is listed here anyway. How can I then renew the coupling to my signal desktop instance?

**Reply by thlkmoeller** [reply]
Hello again,
since I didn't know, what to do, I did the synchronizing process again, as described in my own answer to another thread:

https://github.com/AsamK/signal-cli/discussions/1197

The problems are now:

1) when entering ./signal-cli listDevice, I get 4 linked Desktops, although one of them does now actually not exist anymore due to its lost coupling status
2) on the (again) linked desktop app, there are now all messages till 26th of january and all messages beginning on from 15th of march (today). All messages in between are not there. I know, for whatever reason, the other messages are not synchronized.

And 5 questions:

1) How many days the signal desktop app shall be "not online" in order to loose the linked status, as it happened in my case?
2) Wouldn't it have been possible to bring back to life my desktop app, which was listed in the linked devices (so without doing the coupling process from the beginning) ?
3) How can I anyway synchronize now the messages between end of January and middle of march?
4) What about the old linked device in signal-cli, shall I just leave it there or actively remove it ("removeDevice")?
5) That leads to the last question, how can I exactly identify a device? Can I see in Signal desktop, which device it is? In signal-cli I can see only a long device name consisting of letters and numbers. Can I see the device-names, which I actively chose during the coupling process?

**Reply by thlkmoeller** [reply]
anyone an idea about maybe at least one of the questions...?

**Reply by AsamK** [reply]
2. It's technically possible to reactivate a linked devices, but Signal-Desktop doesn't support that ...
3. That's not how Signal works, old messages are never synchronized. Only new messages will be shown in Signal-Desktop.
4. You can just remove the old linked devices.
5. The device name is encrypted, you need a newer signal-cli version to see the device-name you chose.

**Reply by thlkmoeller** [reply]
a late thank you from me :-)
Does anyone have an answer to question 1?

**Reply by thlkmoeller** [reply]
i guess, i found it myself, by accident.
see here:

https://github.com/signalapp/Signal-Desktop/issues/4328#issuecomment-1329463781
